### ISSUE TITLE
java.lang.AssertionError: Assert failed: (rename? resources) when trying to rename a game object file

### ISSUE THREAD

**Issue by wjl** [defold/defold#9173]
### Expected behaviour

I copied a game object (.go) file and tried to rename it.

### Actual behaviour

An error box popped up. It says:

"An error occurred"

java.lang.AssertionError: Assert failed: (rename? resources)

### Steps to reproduce

I'm making a little test board game, there is only one game object, an atlas, and one collection in the whole brand-new project.

I have a game object file, it just has a sprite and several collision objects, nothing else. It's called "board.go".

I coped it in the editor in the assets panel, and it made "board_copy.go" just fine.

I tried to rename the file in the editor to "piece-1x1.go". This error message pops up, suggesting I report the issue (which is how I got here).

Any time I try to rename the file again, this pops up again, even if I use a different, simple name like "abc" instead of "piece-1x1".

In the console, I get this backtrace:

```
java.lang.AssertionError: Assert failed: (rename? resources)
        at editor.asset_browser$rename.invokeStatic(asset_browser.clj:390)
        at editor.asset_browser$fn__109034$fnk109031_positional__109035.invoke(asset_browser.clj:441)
        at editor.asset_browser$fn__109034$fnk109031__109051$fn__109052.invoke(asset_browser.clj:430)
        at editor.asset_browser$fn__109034$fnk109031__109051.invoke(asset_browser.clj:430)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at editor.handler$invoke_fnk.invokeStatic(handler.clj:256)
        at editor.handler$run.invokeStatic(handler.clj:289)
        at editor.ui$execute_handler_ctx.invokeStatic(ui.clj:1192)
        at editor.ui$execute_command.invokeStatic(ui.clj:1200)
        at editor.ui$run_command.invokeStatic(ui.clj:1440)
        at editor.ui$bind_key_commands_BANG_$reify__14935.handle(ui.clj:1517)
        at com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(CompositeEventHandler.java:321)
        at com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(CompositeEventHandler.java:98)
        at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:219)
        at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:178)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(CompositeEventDispatcher.java:43)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:52)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Scene.processKeyEvent(Scene.java:2197)
        at javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2718)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:430)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
        at com.sun.glass.ui.View.handleKeyEvent(View.java:542)
        at com.sun.glass.ui.View.notifyKey(View.java:966)
        at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
        at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$10(GtkApplication.java:264)
        at java.base/java.lang.Thread.run(Thread.java:833)

```

<hr/>

<table>
<tr><td>Build time</td><td>2024-06-25T16:24:14.962363</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>3642fa5b700b8d6e74e3f0595e75cd50eff54459</td></tr>
<tr><td>Defold engine sha</td><td>d6882f432beca85d460ec42497888157c356d058</td></tr>
<tr><td>Defold version</td><td>1.9.0</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"23afd08de0644b279b8eb7dd7345373a"'>23afd08de0644b279b8eb7dd7345373a</a></td></tr>
<tr><td>GPU</td><td>AMD Radeon Graphics (radeonsi, renoir, LLVM 18.1.6, DRM 3.57, 6.9.7-1-default)</td></tr>
<tr><td>GPU Driver</td><td>4.6 (Compatibility Profile) Mesa 24.1.2</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Linux</td></tr>
<tr><td>OS version</td><td>6.9.7-1-default</td></tr>
<table>

**Comment by wjl** [comment]
Renaming the file outside of the Defold editor works just fine, no further errors occur once this is done, and everything seems to be working fine after that.

**Comment by AGulev** [comment]
duplicate of https://github.com/defold/defold/issues/9134 (but with more info)

**Comment by vlaaad** [comment]
Hi, please check the editor for update, the issue is most probably resolved. Feel free to reopen if the issue returns :)
